### ISSUE TITLE
DataGrid > GridToolbarExport > disableToolbarButton in printOptions doesn't work

### ISSUE THREAD

**Issue by bsides** [mui-org/material-ui-x#3760]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

TL;DR: The option to toggle the print option under `Export` toolbar isn't working.

Both `csvOptions` and `printOptions` share this `GridExportDisplayOptions` interface:

```typescript
interface GridToolbarExportProps extends ButtonProps {
    csvOptions?: GridCsvExportOptions & GridExportDisplayOptions;
    printOptions?: GridPrintExportOptions & GridExportDisplayOptions;
}
```

Which in turn is described like this:

```typescript
interface GridExportDisplayOptions {
    /**
     * If `true`, this export option will be removed from the GridToolbarExport menu.
     * @default false
     */
    disableToolbarButton?: boolean;
}
```

But right now it removes only the option to export into CSV, doing nothing for `Print`. The bug is here. I know it's not yet documented so I'm unsure if it's really a bug but since it's in the code I was expecting to see it working. The example case was taken originally from the documentation, augmenting only those 2 options.

### Expected behavior 🤔

Should hide or show Print option under `Export` toolbar based on this property, default `true`.

### Steps to reproduce 🕹

Steps:

1. Render a DataGrid component
2. Give a custom toolbar component
3. Give props to the export component "printOptions" with the object `{ disableToolbarButton: false }`
4. It should hide the option "Print" under "Export" button, but it doesn't
5. Do the same with "csvOptions", with the same object.
6. You'll notice it hides this option under "Export" button successfully.

### Context 🔦

Was trying to hide the "Print" option under "Export" button in the toolbar of the DataGrid component.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
System:
    OS: macOS 12.1
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.0/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.0/bin/npm
  Browsers:
    Chrome: 97.0.4692.99
    Edge: Not Found
    Firefox: 93.0
    Safari: 15.2
  npmPackages:
    @emotion/react: ^11.6.0 => 11.7.1
    @emotion/styled: ^11.6.0 => 11.6.0
    @mui/base:  5.0.0-alpha.65
    @mui/icons-material: ^5.2.4 => 5.3.0
    @mui/lab: ^5.0.0-alpha.59 => 5.0.0-alpha.65
    @mui/material: ^5.2.2 => 5.3.0
    @mui/private-theming:  5.3.0
    @mui/styled-engine:  5.3.0
    @mui/system:  5.3.0
    @mui/types:  7.1.0
    @mui/utils:  5.3.0
    @mui/x-data-grid: ^5.0.1 => 5.2.2
    @types/react: ^17.0.0 => 17.0.38
    react: ^17.0.2 => 17.0.2
    react-dom: ^17.0.2 => 17.0.2
    typescript: ^4.1.2 => 4.5.4
```
</details>

**Comment by alexfauquette** [comment]
This part of the code is documented here: 
https://mui.com/components/data-grid/export/#remove-the-export-button

I made an example with the 4 configurations which one has a problem in your code?
https://codesandbox.io/s/exportselectorgrid-material-demo-forked-dslz2?file=/demo.js

**Comment by bsides** [comment]
I'm sorry, do you know if this is a recent implementation? Right now it doesn't work in 5.0.1

**Comment by flaviendelangle** [comment]
@bsides It was released in version [5.2.0](https://github.com/mui-org/material-ui-x/blob/master/CHANGELOG.md#muix-data-gridv520--muix-data-grid-prov520)

**Comment by alexfauquette** [comment]
@bsides It has been added in [v5.2.0](https://github.com/mui-org/material-ui-x/releases/tag/v5.2.0)

**Comment by bsides** [comment]
Thank you @alexfauquette and sorry for opening this.

**Comment by alexfauquette** [comment]
No problem. Don't hesitate to open issues if some parts of the doc are unclear, such as we can improve it :)
